cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1677,9 +1703,8 @@ private Map<ProgramRunId, RunRecordDetail> getCompletedRuns(Set<ProgramRunId> pr
     List<List<Field<?>>> allKeys = new ArrayList<>();
     for (ProgramRunId programRunId : programRunIds) {
       // Get all keys without version
-      List<Field<?>> keysWithoutVersion = getRunRecordProgramPrefix(TYPE_RUN_RECORD_COMPLETED,
-                                                                    programRunId.getParent(),
-                                                                    false);
+      List<Field<?>> keysWithoutVersion = getRunRecordProgramReferencePrefix(TYPE_RUN_RECORD_COMPLETED,
+                                                                    programRunId.getParent().getProgramReference());
```

Comment:
indentation

Reply:
if this is only supposed to be getting the latest, it should be taking a `Set<ProgramReference>` as the input argument instead of `Set<ProgramId>`

Reply:
Input is Set<ProgramRunId>, can't use ProgramReference because we need the uuid runId to identify a run